ria_remote/remote.py

```diff
@@ -677,12 +677,27 @@ def transfer_store(self, key, filename):
         dsobj_dir, archive_path, key_path = self._get_obj_location(key)
         key_path = dsobj_dir / key_path
         self.io.mkdir(key_path.parent)
+
         # we need to copy to a temp location to let
         # checkpresent fail while the transfer is still in progress
-        tmp_path = key_path.with_suffix(key_path.suffix + '._')
-        self.io.put(filename, tmp_path)
-        # copy done, atomic rename to actual target
-        self.io.rename(tmp_path, key_path)
+        # and furthermore not interfere with administrative tasks in annex/objects
+        git_dir, _, _ = self.get_layout_locations(self.objtree_base_path, self.archive_id, key)
+        transfer_dir = git_dir / "ria-remote" / "transfer"
+        self.io.mkdir(transfer_dir)
+        tmp_path = transfer_dir / key
+
+        if tmp_path.exists():
+            # Just in case - some parallel job could already be writing to it
+            # at least tell the conclusion, not just some obscure permission error
+            raise RemoteError('{}: upload already in progress'.format(filename))
```

Comment:
Doesn't this effectively prevent any parallel uploads (ie. `git annex copy --jobs 1+`)? I don't think we want that. If needed, we can use the local uuid of the ria remote to prevent collisions. It is globally unique.

Reply:
I don't quite get that. This is not about the transfer dir, but the actual file we want to write to.
We can't write to that in parallel anyway. And if it's parallel chunks they have different keys.

Reply:
Ah, true. My bad, sorry!

In any case, there could still be simultaneous transfers coming from multiple remotes. Maybe we should do:

```
transfer_dir = git_dir / "ria-transfer-{}".format(self.annex.get_uuid())
```

and cache the UUID in `initremote()` for speed.

Reply:
Agree. I built then, with different remotes uploading in mind, so we wouldn't interfere with an ongoing upload. But hten we wouldn't know whether or not that other upload finished sucessfully. So - yes, that's better. Will do. (FTR: `prepare` instead of `initremote`)